### PR TITLE
finishes the controllers model migration

### PR DESCRIPTION
**Pull request by ThyWoof**

### CODE REVIEW

File: SolastaCommunityExpansion/Models/PlayerControllerContext.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        private static bool PlayerInControlOfEnemy { get; set; }
+
+        internal static readonly string[] Controllers = new string[] { "Human", "Computer" };
+
+        internal static readonly int[] ControllersChoices = new int[Settings.MAX_PARTY_SIZE];
+
+        internal static bool IsOffGame => Gui.Game == null;
+
+        internal static bool IsMultiplayer => ServiceRepository.GetService<INetworkingService>().IsMultiplayerGame;
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/PlayerControllerContext.cs:21]: Is it safe to call ServiceRepository.GetService from a static method/property?

Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/PlayerControllerContext.cs:21]: e.g. one the game is started could the network service change from null to not null or vice versa?

Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/PlayerControllerContext.cs:21]: Never mind - it's always available

File: SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs [hunk 2]
```diff
@@ -1,18 +1,18 @@
 ﻿using System.Diagnostics.CodeAnalysis;
 using HarmonyLib;
 
-namespace SolastaCommunityExpansion.Patches.HeroController
+namespace SolastaCommunityExpansion.Patches.PlayerController
 {
     //
-    // TODO: double-check if this indeed required on latest game version. The test scenario is a hero dies when under AI control.
+    // TODO: double-check if this indeed required on latest game version. The test scenario is a hero dies when under AI control or an enemy dies when under player control
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs:7]: Are you double checking?

**Review by PhilPJL — CHANGES_REQUESTED** [review]
need to complete your TODO: or remove

File: SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs (hunk 2 above)
Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs:7]: I'll once I have a chance but as far as I can remember this patch should be there...

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs:7]: when I set logs the sequence is:

1. turn_start
2. optional kill_character
3. turn_end or turn_victory

these patches are here to support these scenarios:

1. Gamers doing a PvP Arena battle. This is when you can select Heroes from Encounters > Characters Pool and Spawn them in an encounter. When I wrote these game had a logic to game over if a hero dies. any hero.
2. What I'm not sure here is if I also need to call stop on heroes controlled by computer.

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/PlayerController/GameLocationCharacterManagerPatcher.cs:7]: 2) isn't a problem as both Stop implementations are safe to call multiple times in sequence. when that happens state stays the same.

**Review by PhilPJL — APPROVED** [review]
Leaving TODO for now

**Pull request merged**